manual_tests/lib/gh/pr.py

```diff
@@ -190,10 +191,36 @@ def check(
 
         return CheckFilter(self, workflow, check_name)
 
+    def toggle_draft(self) -> datetime:
+        since = now()
+        if self.draft:
+            sh.run(("gh", "pr", "ready", self.url))
+        else:
+            sh.run(("gh", "pr", "ready", "--undo", self.url))
+        return since
+
     @classmethod
     def from_branch(cls, branch: Branch, since: datetime) -> Self:
         url = sh.stdout(("gh", "pr", "view", branch, "--json", "url"))
-        return cls(branch, url, since)
+        draft_str = sh.stdout(
+            (
+                "gh",
+                "pr",
+                "view",
+                branch,
+                "--json",
+                "isDraft",
+                "--jq",
+                ".isDraft",
+            )
+        )
+        if draft_str == "true":
+            draft = True
+        elif draft_str == "false":
+            draft = False
+        else:
+            raise ValueError(f"Unknown value for draft: {draft_str}")
```

Comment:
What you've done here is made a parser for a very small subset of json. Let's use the json parser instead:

```suggestion
        draft = sh.json(
            (
                "gh",
                "pr",
                "view",
                branch,
                "--json",
                "isDraft",
                "--jq",
                ".isDraft",
            )
        )
        assert isinstance(draft, bool)
```

Reply:
Ah, cool, I'll have to look more carefully at the available functions in `sh`.